libdevcrypto/Blake2.cpp

```diff
@@ -166,11 +142,10 @@ bytes blake2FCompression(uint32_t _rounds, bytesConstRef _stateVector, bytesCons
         BOOST_THROW_EXCEPTION(InvalidInputSize());
 
     blake2b_state s{};
-    for (size_t i = 0; i < 8; ++i)
-        s.h[i] = load64(&_stateVector[i * sizeof(uint64_t)]);
+    std::memcpy(&s.h, _stateVector.data(), _stateVector.size());
 
-    s.t[0] = load64(_t0.data());
-    s.t[1] = load64(_t1.data());
+    s.t[0] = *reinterpret_cast<uint64_t const*>(_t0.data());
```

Comment:
These should stay `load64` because otherwise this is undefined behavior if the memory is not aligned to 8 bytes.

Reply:
I don't think so.